Hamoye Internship Stage A data analysis

In [1]:
#import necessary libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sins

In [2]:
#First step is to load the dataset 
data_url = "https://raw.githubusercontent.com/WalePhenomenon/climate_change/master/fuel_ferc1.csv"

#load csv file into data variable
data = pd.read_csv(data_url)

#using pd.Dataframe.head() to 
#preview the first 5 rows of the 
#dataset
data.head()

record_id  ...  fuel_cost_per_mmbtu
0   f1_fuel_1994_12_1_0_7  ...                1.121
1  f1_fuel_1994_12_1_0_10  ...                1.120
2   f1_fuel_1994_12_2_0_1  ...                1.650
3   f1_fuel_1994_12_2_0_7  ...                1.970
4  f1_fuel_1994_12_2_0_10  ...                2.570

[5 rows x 11 columns]

In [3]:
#using pandas describe function to
#get a summary of the dataset
data.describe

<bound method NDFrame.describe of                      record_id  ...  fuel_cost_per_mmbtu
0        f1_fuel_1994_12_1_0_7  ...                1.121
1       f1_fuel_1994_12_1_0_10  ...                1.120
2        f1_fuel_1994_12_2_0_1  ...                1.650
3        f1_fuel_1994_12_2_0_7  ...                1.970
4       f1_fuel_1994_12_2_0_10  ...                2.570
...                        ...  ...                  ...
29518  f1_fuel_2018_12_12_0_13  ...                9.030
29519   f1_fuel_2018_12_12_1_1  ...                6.950
29520  f1_fuel_2018_12_12_1_10  ...                8.990
29521  f1_fuel_2018_12_12_1_13  ...                1.110
29522  f1_fuel_2018_12_12_1_14  ...               11.680

[29523 rows x 11 columns]>

It was observed that the dataset contains 29523 rows of data on fuel consumption.

In [4]:
#using the info function to further 
#gain understanding of the dataset
data.info

<bound method DataFrame.info of                      record_id  ...  fuel_cost_per_mmbtu
0        f1_fuel_1994_12_1_0_7  ...                1.121
1       f1_fuel_1994_12_1_0_10  ...                1.120
2        f1_fuel_1994_12_2_0_1  ...                1.650
3        f1_fuel_1994_12_2_0_7  ...                1.970
4       f1_fuel_1994_12_2_0_10  ...                2.570
...                        ...  ...                  ...
29518  f1_fuel_2018_12_12_0_13  ...                9.030
29519   f1_fuel_2018_12_12_1_1  ...                6.950
29520  f1_fuel_2018_12_12_1_10  ...                8.990
29521  f1_fuel_2018_12_12_1_13  ...                1.110
29522  f1_fuel_2018_12_12_1_14  ...               11.680

[29523 rows x 11 columns]>

In [5]:
#checking for missing number
data.isna().sum()

record_id                         0
utility_id_ferc1                  0
report_year                       0
plant_name_ferc1                  0
fuel_type_code_pudl               0
fuel_unit                       180
fuel_qty_burned                   0
fuel_mmbtu_per_unit               0
fuel_cost_per_unit_burned         0
fuel_cost_per_unit_delivered      0
fuel_cost_per_mmbtu               0
dtype: int64

In [6]:
#fuel unit has lots of missing numbers
#we can check the fraction of the missing to the total data
print(data.fuel_unit.isna().sum()/len(data.fuel_unit) *100)

0.609694136774718


From the result, it was noted that the amount of missing rows is so low and removing the affected rows won't affect our analysis.

Secondly, replacing the values would either influence our findings positively or negatively.

So let's drop them for now into a new variable called data_df

In [7]:
data_df = data.dropna()
data_df

record_id  ...  fuel_cost_per_mmbtu
0        f1_fuel_1994_12_1_0_7  ...                1.121
1       f1_fuel_1994_12_1_0_10  ...                1.120
2        f1_fuel_1994_12_2_0_1  ...                1.650
3        f1_fuel_1994_12_2_0_7  ...                1.970
4       f1_fuel_1994_12_2_0_10  ...                2.570
...                        ...  ...                  ...
29518  f1_fuel_2018_12_12_0_13  ...                9.030
29519   f1_fuel_2018_12_12_1_1  ...                6.950
29520  f1_fuel_2018_12_12_1_10  ...                8.990
29521  f1_fuel_2018_12_12_1_13  ...                1.110
29522  f1_fuel_2018_12_12_1_14  ...               11.680

[29343 rows x 11 columns]

In [8]:
#now to verify the drop
data_df.isna().sum()

record_id                       0
utility_id_ferc1                0
report_year                     0
plant_name_ferc1                0
fuel_type_code_pudl             0
fuel_unit                       0
fuel_qty_burned                 0
fuel_mmbtu_per_unit             0
fuel_cost_per_unit_burned       0
fuel_cost_per_unit_delivered    0
fuel_cost_per_mmbtu             0
dtype: int64

In [9]:
#lets check the columns of the dataset
for column in data_df.columns:
    print("Column name:", column)
    print("Datatype", data_df[column].dtypes)
    print()

Column name: record_id
Datatype object

Column name: utility_id_ferc1
Datatype int64

Column name: report_year
Datatype int64

Column name: plant_name_ferc1
Datatype object

Column name: fuel_type_code_pudl
Datatype object

Column name: fuel_unit
Datatype object

Column name: fuel_qty_burned
Datatype float64

Column name: fuel_mmbtu_per_unit
Datatype float64

Column name: fuel_cost_per_unit_burned
Datatype float64

Column name: fuel_cost_per_unit_delivered
Datatype float64

Column name: fuel_cost_per_mmbtu
Datatype float64



Focusing on the columns that holds information that we can work with.
Let carryout out some visualization of the data